crates/zizmor/src/audit/bot_conditions.rs

```diff
@@ -200,11 +215,114 @@ impl BotConditions {
+        // Only emit a patch if the step/job has an `if` key
+        if let Some(If::Expr(expr)) = &step.r#if {
+            Some(Fix {
+                title: "Replace github.actor with github.event.pull_request.user.login".into(),
+                description: "Replace github.actor with github.event.pull_request.user.login to ensure the job runs as the PR author".into(),
+                _key: step.location().key,
+                patches: vec![Patch {
+                    route: step.route().with_keys(&["if".into()]),
+                    operation: Op::Replace(serde_yaml::Value::String(
+                        expr.replace("github.actor", "github.event.pull_request.user.login")
+                            .replace("GitHub.actor", "github.event.pull_request.user.login")
+                            .replace("GitHub.ACTOR", "github.event.pull_request.user.login")
+                            .replace("github.triggering_actor", "github.event.pull_request.user.login")
+                            .replace("GitHub.triggering_actor", "github.event.pull_request.user.login")
+                            .replace("GitHub.TRIGGERING_ACTOR", "github.event.pull_request.user.login")
+                            .into()
+                    )),
+                }],
+            })
+        } else {
+            None
+        }
+    }
+
+    /// Create a fix that suggests changing from pull_request_target to pull_request
+    fn create_trigger_change_fix<'doc>(job: &impl JobExt<'doc>) -> Fix<'doc> {
+        Fix {
+            title: "Change trigger from pull_request_target to pull_request".into(),
+            description: "Change the workflow trigger from pull_request_target to pull_request to prevent running with elevated permissions".into(),
+            _key: job.location().key,
+            patches: vec![Patch {
+                route: job.parent().location().route.with_keys(&["on".into()]),
+                operation: Op::Replace({
+                    let mut map = serde_yaml::Mapping::new();
+                    map.insert(
+                        serde_yaml::Value::String("pull_request".into()),
+                        serde_yaml::Value::Mapping(Default::default()),
+                    );
+                    serde_yaml::Value::Mapping(map)
+                }),
+            }],
+        }
+    }
+
+    /// Create a fix that removes the bot condition entirely
+    fn create_remove_condition_fix<'doc>(step: &Step<'doc>) -> Option<Fix<'doc>> {
+        if step.r#if.is_some() {
+            Some(Fix {
+                title: "Remove bot condition".into(),
+                description: "Remove the bot condition to allow the job to run for all PRs".into(),
+                _key: step.location().key,
+                patches: vec![Patch {
+                    route: step.route().with_keys(&["if".into()]),
+                    operation: Op::Remove,
+                }],
+            })
+        } else {
+            None
+        }
+    }
```

Comment:
I might be misunderstanding the intent here, but I don't _think_ we want to propose removing the entire check here 😅 -- doing so would potentially make some workflows more exploitable, since e.g. an automerge workflow without any actor check would allow _any_ triggerer to induce a merge.